src/storage/src/source/generator/clock.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use mz_ore::now::NowFn;
+use mz_repr::{Datum, Row};
+use mz_storage_types::sources::load_generator::{Event, Generator};
+use mz_storage_types::sources::MzOffset;
+use std::iter;
+use std::time::Duration;
+
+pub struct Clock {
+    pub interval: Duration,
+}
+
+impl Generator for Clock {
+    fn by_seed(
+        &self,
+        now: NowFn,
+        _seed: Option<u64>,
+        resume_offset: MzOffset,
+    ) -> Box<dyn Iterator<Item = (usize, Event<Option<MzOffset>, (Row, i64)>)>> {
+        let tick_ms: u64 = self.interval.as_millis().try_into().unwrap();
```

Comment:
If you're going to unwrap here, it feels like it might be simpler to just have `Clock` store `tick_ms: u64` directly?